libs/execution_context/ebpf_maps.c

```diff
@@ -931,18 +1050,43 @@ ebpf_map_update_entry(
     _In_reads_(value_size) const uint8_t* value,
     int flags)
 {
+    uint32_t cpu_id = 0;
+
     if (!(flags & EBPF_MAP_FLAG_HELPER) && (key_size != map->ebpf_map_definition.key_size)) {
         return EBPF_INVALID_ARGUMENT;
     }
 
-    if (!(flags & EBPF_MAP_FLAG_HELPER) && (value_size != map->ebpf_map_definition.value_size)) {
+    if (!(flags & EBPF_MAP_FLAG_HELPER) &&
+        (value_size != _ebpf_core_map_instance_count(map) * map->ebpf_map_definition.value_size)) {
         return EBPF_INVALID_ARGUMENT;
     }
 
     if (ebpf_map_function_tables[map->ebpf_map_definition.type].update_entry == NULL) {
         return EBPF_INVALID_ARGUMENT;
     }
-    return ebpf_map_function_tables[map->ebpf_map_definition.type].update_entry(map, key, value);
+
+    ebpf_get_current_cpu(&cpu_id);
+
+    // Insert key + zero value into all per-cpu maps if needed.
+    // This is a no-op if the value already exists.
```

Comment:
Why do you need to touch all per-cpu maps?  I don't follow.

Reply:
It's a requirement of the bpf_update_elem.